### PR TITLE
Convert SignalR projects to build with ProjectReference

### PR DESCRIPTION
**Pull request by natemcmaster**
Part of #4246 

Changes:
* Update source code layout to follow the new conventions for this repo
* Update project files to use `<Reference>`
* Update targets to build NPM packages

Reviewers: it's often easier to see the layout changes by looking at the code itself: https://github.com/natemcmaster/AspNetCore/tree/signalr-refs/src/SignalR

### CODE REVIEW

**Comment by mikaelm12** [comment]
>Update project files to use 

...

**Comment by natemcmaster** [comment]
Update projects to use (invisible xml tags)

**Review by JunTaoLuo — APPROVED** [review]
I'm not too familiar with the SignalRisms but the project layout and build/eng updates look good. :shipit: if it builds and tests pass.

File: eng/ProjectReferences.props [hunk 1]
```diff
@@ -124,5 +124,16 @@
     <ProjectReferenceProvider Include="Microsoft.AspNetCore.Authentication.AzureADB2C.UI" ProjectPath="$(RepositoryRoot)src\Azure\AzureAD\Authentication.AzureADB2C.UI\src\Microsoft.AspNetCore.Authentication.AzureADB2C.UI.csproj" />
     <ProjectReferenceProvider Include="Microsoft.AspNetCore.AzureAppServices.HostingStartup" ProjectPath="$(RepositoryRoot)src\Azure\AzureAppServices.HostingStartup\src\Microsoft.AspNetCore.AzureAppServices.HostingStartup.csproj" />
     <ProjectReferenceProvider Include="Microsoft.AspNetCore.AzureAppServicesIntegration" ProjectPath="$(RepositoryRoot)src\Azure\AzureAppServicesIntegration\src\Microsoft.AspNetCore.AzureAppServicesIntegration.csproj" />
+    <ProjectReferenceProvider Include="Microsoft.AspNetCore.SignalR.Client.Core" ProjectPath="$(RepositoryRoot)src\SignalR\clients\csharp\Client.Core\src\Microsoft.AspNetCore.SignalR.Client.Core.csproj" />
+    <ProjectReferenceProvider Include="Microsoft.AspNetCore.SignalR.Client" ProjectPath="$(RepositoryRoot)src\SignalR\clients\csharp\Client\src\Microsoft.AspNetCore.SignalR.Client.csproj" />
```
[the remark this answers is not in the document]
Comment (natemcmaster) [reply eng/ProjectReferences.props]: Naming is hard. I figured its presence under src/SignalR/ was sufficient, but I don't care either way. If there is something you'd prefer instead, lmk.

**Comment by natemcmaster** [comment]
Updated to react to dropping netcoreapp2.0 test TFMs. I'd like to merge this later today. PTAL.

**Comment by natemcmaster** [comment]
I think something in the SignalR tests is causing the Linux builds to run out of memory, or interfering with the network connection between agent and host. Back when SignalR was in its own repo, did you run into problems like this?

**Comment by BrennanConroy** [comment]
Nope, never seen that

**Comment by natemcmaster** [comment]
About to push a big update to this PR. At least one of the problems we have now is that the repo targets implemented to run npm pack always run, even if you are using the build.cmd script in a different subfolder.

The fix for this is to replace those targets with a shim, like a .npmproj file, which invokes npm build/test/ci etc.

To unblock this, we need: https://github.com/aspnet/BuildTools/pull/917

**Comment by natemcmaster** [comment]
Updated BuildTools. Should be unblocked now.

**Review by BrennanConroy — APPROVED** [review]
Don't know what's up with the build

**Comment by natemcmaster** [comment]
I'll handle kestrel's test failures separately.

**Pull request merged**